src/graph/sampler.cc

```diff
@@ -88,12 +89,19 @@ class ArrayHeap {
   /*
    * Sample a vector by given the size n
    */
-  void SampleWithoutReplacement(size_t n, std::vector<size_t>* samples) {
+  size_t SampleWithoutReplacement(size_t n, std::vector<size_t>* samples) {
     // sample n elements
-    for (size_t i = 0; i < n; ++i) {
+    size_t i = 0;
+    for (; i < n; ++i) {
+      // heap is empty
+      if (heap_[1] == 0) {
+        break;
```

Comment:
what does this mean?

Reply:
when use SampleWithoutReplacement, node will be deleted (weight cleaned to 0). So when the heap_[1] == 0, the whole heap is cleaned (no node left)

Reply:
so there was a bug in the previous implementation?